File: client/src/services/category.js
import { apiService } from "@/api";
import authHeader from "@/services/auth-header";
export const CategoriesAPI = {
  getCategoryCollection: async () => {
    const res = (await apiService.get("/category/", { headers: authHeader() }))
      .data;
    return res;
  },
  createCategory: async (data) => {
    const res = (
      await apiService.post("/category/", data, { headers: authHeader() })
    ).data;
    return res;
  },
};

File: api-db/app/apis/category.py
from flask_restx import Namespace, Resource, fields
from flask import request
from ..services.category import CategoryService

api = Namespace("category", "Category Resource")

category = api.model(
    "Category",
    {
        "id": fields.Integer,
        "name": fields.String,
    },
)


#  Dummy Data


@api.route("/")
class CategoryCollection(Resource):
    @api.marshal_with(category, envelope="categories")
    def get(self):
        return CategoryService.get_categories()

    @api.marshal_with(category, envelope="category")
    def post(self):
        payload = request.json
        return CategoryService.create_category(payload)
